#!/usr/bin/env python
PACKAGE = "simple_bgc"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("configure_flag", bool_t, 0, "Configure Flag", False)
gen.add("profile_id", int_t, 0, "Profile Flag", 0, 0, 2)

gen.add("roll_p", int_t, 0, "Roll P Gain", 0, 0, 255)
gen.add("roll_i", int_t, 0, "Roll I Gain", 0, 0, 255)
gen.add("roll_d", int_t, 0, "Roll D Gain", 0, 0, 255)
gen.add("roll_power", int_t, 0, "Roll Power", 0, 0, 255)
gen.add("roll_invert", bool_t, 0, "Rotation Invert ON/OFF", False)
gen.add("roll_poles", int_t, 0, "Roll Motor Poles", 0, 0, 255)

gen.add("pitch_p", int_t, 0, "Pitch P Gain", 0, 0, 255)
gen.add("pitch_i", int_t, 0, "Pitch I Gain", 0, 0, 255)
gen.add("pitch_d", int_t, 0, "Pitch D Gain", 0, 0, 255)
gen.add("pitch_power", int_t, 0, "Pitch Power", 0, 0, 255)
gen.add("pitch_invert", bool_t, 0, "Rotation Invert ON/OFF", False)
gen.add("pitch_poles", int_t, 0, "Pitch Motor Poles", 0, 0, 255)

gen.add("baud", int_t, 0, "Buadrate", 0, 0, 921600)
gen.add("motor", bool_t, 0, "Motor ON/OFF", True)

exit(gen.generate(PACKAGE, "simple_bgc", "Control"))
